accounts-db/src/accounts_db.rs

```diff
@@ -4341,11 +4341,21 @@ impl AccountsDb {
                 // clean needs to take care of this dead slot
                 self.accounts_index.add_uncleaned_roots([slot]);
             }
-            info!(
+
+            let mut msg = format!(
                 "Unexpected shrink for slot {} alive {} capacity {}, \
-                likely caused by a bug for calculating alive bytes.",
+                    likely caused by a bug for calculating alive bytes.",
                 slot, shrink_collect.alive_total_bytes, shrink_collect.capacity
             );
+            if !shrink_collect.all_are_zero_lamports {
+                // if all are zero lamports, then we expect that we would like to mark the whole slot dead, but we cannot. That's clean's job.
+                msg.push_str(&format!(
+                    "All alive bytes are zero: {}, {}",
+                    store.alive_bytes(),
+                    shrink_collect.zero_lamport_single_ref_pubkeys.len() * aligned_stored_size(0)
+                ));
+            }
+            info!("{}", &msg);
```

Comment:
I don't want any msg if `all_are_zero_lamports`. This is a false positive and expected. We have zero 'alive' bytes, but we have many alive zero lamport single ref accounts. This is normal behavior and will be resolved by `add_uncleaned_roots` above. I think the whole pr here is `if !shrink_collect.all_are_zero_lamports {` around the existing `info`.